app/controllers/api/notifications_controller.rb

```diff
@@ -9,7 +9,8 @@ def index
                                        .by_read_status(status)
                                        .latest_of_each_link
 
-    @notifications = Notification.from(latest_notifications, :notifications) # latest_notifications のクエリで指定している ORDER BY の順序を他と混ぜないようにするため、from を使ってサブクエリとした
+    @notifications = Notification.with_avatar
+                                 .from(latest_notifications, :notifications) # latest_notifications のクエリで指定している ORDER BY の順序を他と混ぜないようにするため、from を使ってサブクエリとした
```

Comment:
```suggestion
    # latest_notifications のクエリで指定している ORDER BY の順序を他と混ぜないようにするため、
    # from を使ってサブクエリとした
    @notifications = Notification.with_avatar
                                 .from(latest_notifications, :notifications) 
```

コメントを後ろに書くのはちょっとした短い1行に対してやるやり方なので、長い行や複数行のコードのまとまりにコメントするならこういう感じがいいかもです。